homeassistant/components/samsungtv/media_player.py

```diff
@@ -179,21 +196,77 @@ async def async_update(self) -> None:
                 STATE_ON if await self._bridge.async_is_on() else STATE_OFF
             )
 
-        if self._attr_state == STATE_ON and not self._app_list_event.is_set():
-            await self._bridge.async_request_app_list()
-            if self._app_list_event.is_set():
-                # The try+wait_for is a bit expensive so we should try not to
-                # enter it unless we have to (Python 3.11 will have zero cost try)
-                return
-            try:
-                await asyncio.wait_for(self._app_list_event.wait(), APP_LIST_DELAY)
-            except asyncio.TimeoutError as err:
-                # No need to try again
-                self._app_list_event.set()
-                LOGGER.debug(
-                    "Failed to load app list from %s: %s", self._host, err.__repr__()
+        if self._attr_state != STATE_ON:
+            return
+
+        startup_tasks: list[Coroutine[Any, Any, None]] = []
+
+        if not self._app_list_event.is_set():
+            startup_tasks.append(self._async_startup_app_list())
+
+        if not self._upnp_device and self._ssdp_rendering_control_location:
+            startup_tasks.append(self._async_startup_upnp())
+
+        if startup_tasks:
+            await asyncio.gather(*startup_tasks)
+
+        if not (service := self._get_upnp_service()):
+            return
+
+        get_volume, get_mute = await asyncio.gather(
+            service.action("GetVolume").async_call(InstanceID=0, Channel="Master"),
```

Comment:
I don't think this is in line with our design requirements. Protocol details should stay in a 3rd party library. The library should handle the UPnP setup and command details.

Reply:
I figured this was ok since we have the similar pattern in the upnp and dlna* integrations

Reply:
Maybe we create a new profile for Samsung Tv 

async_upnp_client.profiles.samsungtv?

Reply:
I would consider them exceptions as they are general integrations similar to mqtt.

The Samsung TV integration is not that.

Reply:
@epenet async_upnp_client.profiles.samsungtv or a new library?  It doesn't seem to fit in any of the existing ones.

Not sure what is better

Reply:
I haven't looked into the details, and I won't be back home for a while so but I wonder if we could actually get subscriptions working as well so we could get push updates.